### PR TITLE
Fix critical csv bugs

### PR DESCRIPTION
**Pull request by friendlymatthew**
This PR contains bug fixes and extra coverage to assert correctness between index files and CSV parsing

### CODE REVIEW

File: pkg/appendable/csv_handler.go [hunk 1]
```diff
@@ -174,7 +188,7 @@ func (i *IndexFile) handleCSVLine(dec *csv.Reader, headers []string, path []stri
 			return fmt.Errorf("unexpected type '%T'", value)
 		}
 
-		cumulativeLength += fieldLength
+		cumulativeLength += fieldLength + 1
```
Comment (friendlymatthew) [inline pkg/appendable/csv_handler.go:191]: account for `,`

File: pkg/appendable/index_file_test.go [hunk 2]
```diff
@@ -61,6 +50,47 @@ func TestIndexFile(t *testing.T) {
 
 	})
 
+	t.Run("compare mock index file after appending", func(t *testing.T) {
```
Comment (friendlymatthew) [inline pkg/appendable/index_file_test.go:53]: we now check for index file equality after appending an extra data line to both formats

File: pkg/appendable/index_file_test.go [hunk 3]
```diff
@@ -69,22 +99,14 @@ func compareIndexRecord(ir1, ir2 *protocol.IndexRecord, fieldType protocol.Field
...
 			return false, fmt.Sprintf("Field Length do not align\ti1: %v, i2: %v", ir1.FieldLength, ir2.FieldLength)
 		}
-	} /* else {
-		if ir1.FieldStartByteOffset != ir2.FieldStartByteOffset {
-			return false, fmt.Sprintf("FieldStartByteOffset do not align\ti1: %v, i2: %v", ir1.FieldStartByteOffset, ir2.FieldStartByteOffset)
-		}
-
-		if ir1.FieldLength != ir2.FieldLength {
+	} else {
+		if ir1.FieldLength != ir2.FieldLength+2 {
```
Comment (friendlymatthew) [inline pkg/appendable/index_file_test.go:106]: account for the "" that `jsonl` has

File: pkg/appendable/csv_handler.go (hunk 1 above)
Comment (kevmo314) [reply pkg/appendable/csv_handler.go:191]: Nice, off by one error :)

**Review by kevmo314 — APPROVED** [review]

**Pull request merged**